### DISCUSSION TITLE
Ideas on MagNet Toolkit

### DISCUSSION THREAD

**Discussion by minjiechen** [upb-lea/mag-net-hub#1] [Ideas]
One recommendation - we can make the core loss prediction function reading both time sequence B(t), and descriptive language, "sine", "trig", "trapz", with specified duty ratios, something like this:

something like this - 
Pv = f ("N87", "sine", f, Bdc, T, "paderborn") - reading descriptive language for waveforms
Pv = f ("N49", "trig", D, Bdc, T, "sydney") - reading descriptive language for waveforms
Pv = f ("N27", B(t), T, "princeton") - reading B(t) for waveforms

perhaps B(t) should be compatible with arbitrary number of steps. 
we will internally reformat/generate B(t) for a particular resolution (say 1024) based on the information we receive.

**Comment by wkirgsn** [comment]
Great ideas, i will implement that. 
Note that the arbitrary b sequence length is in fact already implemented in the Paderborn model, as it needs 1024 samples per period. 
Likewise, other models will certainly have such resampling/Interpolation, too.

**Comment by wkirgsn** [comment]
Hi Minjie, is there a Princeton model that we could add to the Toolkit?

**Comment by minjiechen** [comment]
Not sure, ccing Shukai/Haoran ~

Bests,
Minjie
------------------
Minjie Chen
Princeton University
Zoom: https://princeton.zoom.us/j/6177687869

**Reply by ehavugi** [reply]
related add https://github.com/PrincetonUniversity/magnet/pull/125
